wallet/createtx.go

```diff
@@ -109,148 +109,159 @@ func (s secretSource) GetScript(addr btcutil.Address) ([]byte, error) {
 func (w *Wallet) txToOutputs(outputs []*wire.TxOut, keyScope *waddrmgr.KeyScope,
 	account uint32, minconf int32, feeSatPerKb btcutil.Amount,
 	coinSelectionStrategy CoinSelectionStrategy, dryRun bool) (
-	tx *txauthor.AuthoredTx, err error) {
+	*txauthor.AuthoredTx, error) {
 
 	chainClient, err := w.requireChainClient()
 	if err != nil {
 		return nil, err
 	}
 
-	dbtx, err := w.db.BeginReadWriteTx()
-	if err != nil {
-		return nil, err
-	}
-	defer func() { _ = dbtx.Rollback() }()
-
-	addrmgrNs, changeSource, err := w.addrMgrWithChangeSource(
-		dbtx, keyScope, account,
-	)
-	if err != nil {
-		return nil, err
-	}
-
 	// Get current block's height and hash.
 	bs, err := chainClient.BlockStamp()
 	if err != nil {
 		return nil, err
 	}
 
-	eligible, err := w.findEligibleOutputs(
-		dbtx, keyScope, account, minconf, bs,
-	)
-	if err != nil {
-		return nil, err
-	}
-
-	var inputSource txauthor.InputSource
-
-	switch coinSelectionStrategy {
-	// Pick largest outputs first.
-	case CoinSelectionLargest:
-		sort.Sort(sort.Reverse(byAmount(eligible)))
-		inputSource = makeInputSource(eligible)
-
-	// Select coins at random. This prevents the creation of ever smaller
-	// utxos over time that may never become economical to spend.
-	case CoinSelectionRandom:
-		// Skip inputs that do not raise the total transaction output
-		// value at the requested fee rate.
-		var positivelyYielding []wtxmgr.Credit
-		for _, output := range eligible {
-			output := output
-
-			if !inputYieldsPositively(&output, feeSatPerKb) {
-				continue
-			}
-
-			positivelyYielding = append(positivelyYielding, output)
+	var tx *txauthor.AuthoredTx
+	err = walletdb.Update(w.db, func(dbtx walletdb.ReadWriteTx) error {
```

Comment:
I think there's a (preexisting) race here with the `BlockStamp()` since we may be past the block by the time the transaction commits. It's not related to the `Update` refactor and not sure if it's porblematic.

Reply:
I think that's okay. If a new block comes in while we're in this DB transaction, we just won't "see" any UTXOs in that block. I think with the wallet's own (internal) unique coin lock maybe we can't process new blocks while a wallet operation is ongoing anyway.